### PR TITLE
Enhance error handling for Python client

### PR DESCRIPTION
**Pull request by mellis13**
This PR adds additional error handling to support Python client connection.  Additionally, it fixes an error that was introduced in the previous error handling enhance PR that repeatedly forms a connection for redis cluster.

### CODE REVIEW

File: src/cpp/redis.cpp [hunk 1]
```diff
@@ -356,15 +361,24 @@ inline void Redis::_connect(std::string address_port)
...
                 break;
             }
+            else {
+                run_sleep = true;
+            }
         }
         catch (std::exception& e) {
             if(i == n_trials) {
-                delete this->_redis;
-                this->_redis = 0;
```
Comment (billschereriii) [inline src/cpp/redis.cpp:363]: Leave in the assignment to NULL (and change from 0 to NULL)

File: src/cpp/redis.cpp [hunk 2]
```diff
@@ -356,15 +361,24 @@ inline void Redis::_connect(std::string address_port)
...
             }
             run_sleep = true;
         }
+        catch (...) {
+            if(i == n_trials) {
+                throw std::runtime_error("A non-standard exception "\
+                                         "encountered during client "\
+                                         "connection.");
+            }
+            run_sleep = true;
```
Comment (billschereriii) [inline src/cpp/redis.cpp:380]: Fail back to caller if we hit a weird exception?

File: src/cpp/rediscluster.cpp [hunk 3]
```diff
@@ -470,18 +475,22 @@ inline void RedisCluster::_connect(std::string address_port)
...
             this->_redis_cluster =
                 new sw::redis::RedisCluster(address_port);
-        }
-        catch (std::bad_alloc& e) {
-            throw std::runtime_error(e.what());
+            return;
         }
         catch (std::exception& e) {
-            delete this->_redis_cluster;
-            this->_redis_cluster = 0;
```
Comment (billschereriii) [inline src/cpp/rediscluster.cpp:479]: Leave in assignment to NULL (and switch 0 => NULL)

**Comment by billschereriii** [comment]
PR looks good. A few minor comments above

File: src/cpp/redis.cpp (hunk 1 above)
Comment (mellis13) [reply src/cpp/redis.cpp:363]: I see what you are saying, but I think we should put the assignment to NULL up in teh construction portion, not the PONG portion

File: src/cpp/rediscluster.cpp (hunk 3 above)
Comment (mellis13) [reply src/cpp/rediscluster.cpp:479]: Agreed.

File: src/cpp/redis.cpp (hunk 1 above)
Comment (billschereriii) [reply src/cpp/redis.cpp:363]: Agree, keep assignment with the deletion

**Review by billschereriii — APPROVED** [review]

**Pull request merged**